text/0011-fluent-in-prefs-design-review.md

```diff
@@ -0,0 +1,61 @@
+# Fluent in Prefs Design Review
+
+## Introduction
+
+[Fluent](http://projectfluent.org/) is a "family of localization libraries designed to improve how software is translated." Mozilla's localization engineering team (the "engineering team") is migrating Firefox's localization architecture from DTD/properties files to FTL files processed by Fluent, which will enable richer and more accurate translations, runtime locale loading/switching, runtime LTR/RTL and pseudo-locale switching, and a fallback locale. The first target for this migration is the preferences interface (the about:preferences page and its various subdialogs).
+
+The engineering team met with a set of architects and domain experts (the "review team") to discuss risks and mitigations, which constituted a lightweight "design review" of the project's architecture. This document describes the concerns that the review team raised and the risk mitigations that the teams discussed.
+
+## Why Preferences
+
+The team chose preferences because they're text-heavy, include users' names (which exercises Fluent's support for grammatical gender), and because the precision of text matters in preferences to ensure users clearly understand the implications of changes. So preferences are a good way to prove the change to Fluent before adopting it more widely.
+
+Fluent is a big change to the localization process, and the team plans to validate the change by translating just four strings initially (the "name" strings in brand.dtd/properties?) and then verifying those changes across the localization toolchain.
```

Comment:
Rough suggestion: "Fluent is a big change to the localization process, and the team is rolling out the change in stages. The first stage landed in Firefox 58 and enabled a single string localized using Fluent (https://bugzilla.mozilla.org/show_bug.cgi?id=1402069). Second stage is about to land (https://bugzilla.mozilla.org/show_bug.cgi?id=1411012) and it migrates four strings to FTL. Each of those is meant to test the complete localization toolchain readiness for the new system. After them, the Preferences migration will begin."